spacemeshos/smapp
# Support new transactions and accounts structure

It closes #936 
Still in the testing phase and most likely will need some tweaks.
Some parts are still in the process, but it is hard to implement it without ability to test.

---

### How to test
Follow the instructions in README.md and then follow some extra steps:
1. Build `go-spacemesh` from the same commit that we're using now in the private testnet (now we're using
2. Copy it into `smapp/node/YOUR_OS` (`mac` / `win` / `linux`)
3. Make them executable `chmod 755 ./node/mac/*`
4. Create a new directory somewhere and put it there
    - `networks.json` (see below)
    - `config.json` which content you should take from here (if you have access): https://github.com/spacemeshos/Applications/blob/main/spacemesh-networks/devnet/999/config.yaml
5. Serve them (I am using `python -m SimpleHTTPServer`)
6. Run Smapp with dev env: `DISCOVERY_URL=http://localhost:8000/networks.json yarn start`
7. Create/open your wallet and choose the network from the list

networks.json
```
[
  {
    "netName": "PRIVATE TESTNET",
    "netID": 999,
    "conf": "http://0.0.0.0:8000/config.json",
    "grpcAPI": "https://api-devnet999.spacemesh.dev/",
    "jsonAPI": "https://api-json-devnet999.spacemesh.dev/",
    "explorer": "https://explorer.spacemesh.io/",
    "explorerAPI": "https://explorer-api-devnet225.spacemesh.io/",
    "explorerVersion": "v0.0.10",
    "explorerConf": "https://storage.googleapis.com/spacecraft-data/devnet225-archive/config.json",
    "dash": "https://dash.spacemesh.io/",
    "dashAPI": "wss://dash-api-devnet225.spacemesh.io/ws",
    "dashVersion": "v0.0.2",
    "repository": "spacemeshos/go-spacemesh-dev",
    "minNodeVersion": "d298a63",
    "maxNodeVersion": "d298a63",
    "minSmappRelease": "0.2.6",
    "latestSmappRelease": "0.2.6",
    "smappBaseDownloadUrl": "https://downloads.spacemesh.io",
    "nodeBaseDownloadUrl": "https://downloads.spacemesh.io"
  }
]
```
Don't be confused with most of the values here, the only things that we really need for testing are `netID`, `conf`, `grpcAPI`

---

To simplify testing transactions I also wrote a small CLI-tool that composes and optionally signs the transaction. Afterward you can just copy the byte array of the signed transaction and send it to `TransactionService.SubmitTransaction` GRPC API via your favorite GRPC-client (E.G. `grpcurl`).
Just run `yarn debug:tx` and follow instructions ;)
